src/sentry/workflow_engine/handlers/condition/group_state_handlers.py

```diff
@@ -40,3 +40,18 @@ def evaluate_value(job: WorkflowJob, comparison: Any) -> bool:
         has_escalated = job.get("has_escalated", False)
         is_escalating = has_reappeared or has_escalated
         return is_escalating and job["event"].group.priority == PriorityLevel.HIGH
+
+
+@condition_handler_registry.register(Condition.FIRST_SEEN_EVENT)
+class FirstSeenEventConditionHandler(DataConditionHandler[WorkflowJob]):
+    @staticmethod
+    def evaluate_value(job: WorkflowJob, comparison: Any) -> bool:
+        state = job.get("group_state")
+        if state is None:
+            return False
+
+        workflow = job.get("workflow")
+        if workflow is None or workflow.environment_id is None:
+            return state["is_new"]
+
+        return state["is_new_group_environment"]
```

Comment:
only `job["event"]` is guaranteed to exist, so for other keys you'll have to do `.get()`

Reply:
I thought it was fine to do like this because state is a `TypedDict`? but I used `job.get("group_state")` to get the state

https://github.com/getsentry/sentry/blob/3c38f01e21a3e7064771bad3a19679a929d5a333/src/sentry/eventstream/base.py#L30-L34

Reply:
oh my b, you're right